### ISSUE TITLE
Add test for sequence parallel

### ISSUE THREAD

**Issue by suchenzang** [facebookresearch/metaseq#600]
Right now, we have the option to use sequence parallel via the `--sequence-parallel` flag: https://github.com/facebookresearch/metaseq/blob/a6ef598cc7b4dac394ba2eab5d0e75ca27a9e8c0/metaseq/modules/transformer_decoder_layer.py#L210-L223

Now that https://github.com/facebookresearch/metaseq/issues/578 is completed, we should add a test here to check rough equivalence between going through the sequence-parallel code-path (say, with MP 2) vs the current non sequence-parallel run included in the unit tests for the 8M model.

**Comment by bashnick** [comment]
**Tl;dr**
Currently the test for --sequence-parallel is blocked by the old version of PyTorch (1.9) on CircleCI and we need a new image with cuda 11.6 to update PyTorch.

**Details**
The --sequence-parallel uses the PyTorch torch.distributed._reduce_scatter_base method. This method is introudced starting from PT1.10 and the current version of PT in CircleCI is PT1.9.

I’ve tried in different ways to upgrade PT to PT1.10+cu111 and PT1.13+cu116 on CircleCI with different setups of other libraries, but weren’t able to make Apex installation work - it usually fails with an vague “Command '['ninja', '-v']' returned non-zero exit status 1.”

**Next steps**
I've requested a new image from CircleCI with cuda 11.6. Once we have it (hopefully beginning of Feb 2023), we can update the PyTorch version in the CircleCI config to PT1.13+cu116 and install the dependancies.
